ee/clickhouse/sql/events.py

```diff
@@ -1,12 +1,15 @@
-from ee.kafka_client.topics import KAFKA_EVENTS
-from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE, DEBUG
+from django.conf import settings
 
-from .clickhouse import KAFKA_COLUMNS, REPLACING_MERGE_TREE, STORAGE_POLICY, kafka_engine, table_engine
+from ee.clickhouse.sql.clickhouse import KAFKA_COLUMNS, STORAGE_POLICY, kafka_engine
+from ee.clickhouse.sql.table_engines import Distributed, ReplacingMergeTree, ReplicationScheme
+from ee.kafka_client.topics import KAFKA_EVENTS
 
-EVENTS_TABLE = "events"
+EVENTS_DATA_TABLE = lambda: "sharded_events" if settings.CLICKHOUSE_REPLICATION else "events"
```

Comment:
Nit: it could be nice to keep the same table name regardless of if it's sharded or not